mlir/lib/Dialect/ArmNeon/Transforms/LowerContractionToSMMLAPattern.cpp

```diff
@@ -150,16 +150,30 @@ class LowerContractionToSMMLAPattern
       Value tiledAcc =
           extractOperand(op.getAcc(), accPermutationMap, accOffsets);
 
+      // With vecmat, tiled LHS and ACC will contain only one of 2 necessary
+      // rows along dimM. Broadcast both to the full width
+      if (isVecmat) {
+        auto lhsBroadcastType = VectorType::get(
+            {2, 8}, tiledLhs.getType().cast<ShapedType>().getElementType());
+        tiledLhs = rewriter.create<vector::BroadcastOp>(loc, lhsBroadcastType,
+                                                        tiledLhs);
+        auto accBroadcastType = VectorType::get(
+            {2, 2}, tiledAcc.getType().cast<ShapedType>().getElementType());
+        tiledAcc = rewriter.create<vector::BroadcastOp>(loc, accBroadcastType,
+                                                        tiledAcc);
```

Comment:
@dcaballe I couldn't find an equivalent vector op for `tensor::empty` so using a broadcast here. Half the matmul is not needed so the data can be anything. If there is something simpler, let me know.

Reply:
You can use undef to get an empty vector like this: `  %undef = llvm.mlir.undef : vector<10xi8>`

Then you can use `vector.insert` to fill it.

Reply:
Yes, that would work! I'm not sure if we would have to introduce the LLVM dialect dependence, though. I think in general we've been doing the same but using `arith.constant 0`  instead of the undef.